### ISSUE TITLE
Incorrect transfer of metadata

### ISSUE THREAD

**Issue by JoCat** [savonet/liquidsoap#2478]
**Describe the bug**
Hello, I'm trying to display a broadcast on YouTube and a couple of other services. But I am facing the following problem.

If I use the first method, then everything works, but then optimizations using supported processor instructions are not used when encoding

If method 2, then optimizations are activated, but stream metadata is not correctly transmitted
![image](https://user-images.githubusercontent.com/13483132/176028681-9a967f27-2a22-46a0-8e7e-6538252fd7ac.png)

![image](https://user-images.githubusercontent.com/13483132/176026780-fc3db3e6-b62d-48e8-92a8-ef1695b61c2f.png)
![image](https://user-images.githubusercontent.com/13483132/176028846-9374e5f0-d4c8-47f5-9316-df08b1df8012.png)

**To Reproduce**

*Method №1*
```ocaml
# --- Vars ---

workdir = "/root/liquidsoap_test/"
audio_url = "https://radio.shiza-project.com/ara-ara"
youtube_key = "youtube_key"

# --- Set settings ---

settings.log.file.set(true)
settings.log.file.path.set("#{workdir}liquidsoap_video.log")
settings.log.file.append.set(true)
settings.log.stdout.set(true)
settings.log.level.set(3)
settings.init.allow_root.set(true)
video.frame.rate.set(30)

# --- Processing ---

audio_source = mksafe(input.http(audio_url))
video_source = single("#{workdir}bg.png")

# Mux it with the audio
stream = mux_video(video=video_source, audio_source)

# --- Output ---

encoder = %ffmpeg(format="flv",
    %audio(codec="aac", b="128k", channels=2, samplerate=44100),
    %video(codec="libx264", b="4000k", width=1280, height=720, coder="ac", preset="ultrafast", "x264-params"="min-keyint=100:keyint=100"),
)

output.youtube.live.rtmp(
    key = youtube_key,
    encoder = encoder,
    stream
)
```

*Method №2*
```ocaml
# --- Vars ---

workdir = "/root/liquidsoap_test/"
audio_url = "https://radio.shiza-project.com/ara-ara"
youtube_key = "youtube_key"

# --- Set settings ---

settings.log.file.set(true)
settings.log.file.path.set("#{workdir}liquidsoap_video.log")
settings.log.file.append.set(true)
settings.log.stdout.set(true)
settings.log.level.set(3)
settings.init.allow_root.set(true)
video.frame.rate.set(30)

# --- Processing ---

audio_source = mksafe(input.http(audio_url))
video_source = single("#{workdir}bg.png")

# Mux it with the audio
stream = mux_video(video=video_source, audio_source)

stream = ffmpeg.encode.audio_video(
    %ffmpeg(
        %audio(codec="aac", b="128k", channels=2, samplerate=44100),
        %video(codec="libx264", b="4000k", width=1280, height=720, coder="ac", preset="ultrafast", "x264-params"="min-keyint=100:keyint=100"),
    ),
    stream
)

# --- Output ---
encoder = %ffmpeg(format="flv",
    %audio.copy,
    %video.copy,
)

output.youtube.live.rtmp(
    key = youtube_key,
    encoder = encoder,
    stream
)
```

**Expected behavior**
Using optimizations on method 1 or correct return of metadata on method 2.

**Version Information**
   - OS: Debian 11
   - Version: 2.0.5

**Installation method**
opam

**Comment by toots** [comment]
Hi! Thanks for the report. What do you mean by stream metadata? Usually it means artist/song/title, is that the case here?

**Comment by JoCat** [comment]
> Hi! Thanks for the report. What do you mean by stream metadata? Usually it means artist/song/title, is that the case here?

information about the video stream, its resolution in this case

**Comment by toots** [comment]
As displayed by youtube?

**Comment by JoCat** [comment]
> As displayed by youtube?

Yes. In Telegram, in this case, the stream came in fragments. Haven't tested other services.

**Comment by toots** [comment]
Gotcha. Are you sending the resulting stream to multiple destinations? Otherwise, for one output only, the two methods should provide the same CPU usage.

Hopefully we can fix that soon.

**Comment by JoCat** [comment]
> Are you sending the resulting stream to multiple destinations?

Yes

> Otherwise, for one output only, the two methods should provide the same CPU usage.

Actually yes

**Comment by toots** [comment]
Wait no, it just did!

**Comment by toots** [comment]
<img width="1091" alt="Screen Shot 2022-07-14 at 10 04 18 AM" src="https://user-images.githubusercontent.com/871060/179014466-3d255bb7-a66f-4cc8-87f8-b56748d47b37.png">

**Comment by toots** [comment]
Ok I was able to reproduce and figure it out. It's a tricky corner case of inline encoders. I have added the following to the [FFmpeg support documentation](https://www.liquidsoap.info/doc-dev/ffmpeg.html):

> `liquisoap` provides operators to encode data using `%ffmpeg` and re-use it accross output. This is called _inline encoding_. Here's an example:
> 
> ```liquidsoap
> audio_source = single(audio_url)
> video_source = single(image)
> 
> stream = mux_video(video=video_source, audio_source)
> 
> stream = ffmpeg.encode.audio_video(
>     %ffmpeg(
>         %audio(codec="aac", b="128k"),
>         %video(codec="libx264", b="4000k")
>     ),
>     stream
> )
> 
> flv = %ffmpeg(
>     format="flv",
>     %audio.copy,
>     %video.copy,
> )
> 
> # Send to one youtube output:
> output.youtube.live.rtmp(
>     encoder = flv,
>     stream,
>     ...
> )
> 
> mpegts = %ffmpeg(
>     format="mpegts",
>     %audio.copy,
>     %video.copy,
> )
> 
> # And to a hls one:
> output.file.hls(
>   ["mpegts", mpegts],
>   stream,
>   ...
> )
> ```
> 
> Working with encoded data, however, requires a bit of knowledge of ffmpeg internal and media codecs and containers. Here, for instance, this stream
> will have issues because the `flv` format requires global data, something that in ffmpeg terms is called `extradata`.
> 
> When working with a single encoder such as:
> 
> ```liquidsoap
> %ffmpeg(
>   format="flv",
>   %audio(codec="aac", b="128k"),
>   %video(codec="libx264", b="4000k")
> )
> ```
> 
> We are aware when initializing the encoders that it is aimed for a `flv` container so the code implicitely enables the global header for each encoder.
> 
> However, when encoding inline, we do not know at the time of encoding the container that will be used to encapsulate the stream, even worst, it can be
> used potentially with different containers with different requirements!
> 
> In our case here, you have two ways to solve the issue:
> 
> If you know that all the containers will be okay with global header, you can enable the corresponding flag in the encoder:
> 
> ```liquidsoap
> stream = ffmpeg.encode.audio_video(
>     %ffmpeg(
>         %audio(codec="aac", b="128k", flags="+global_header"),
>         %video(codec="libx264", b="4000k", flags="+global_header")
>     ),
>     stream
> )
> ```
> 
> However, it is also possible that one stream needs global header but not the other one, which is the case here with `mpegts`. In this case, you can
> use the _bitstream filter_ `ffmpeg.filter.bitstream.extract_extradata` to extract global data to only one stream:
> 
> ```
> audio_source = single(audio_url)
> video_source = single(image)
> 
> stream = mux_video(video=video_source, audio_source)
> 
> stream = ffmpeg.encode.audio_video(
>     %ffmpeg(
>         %audio(codec="aac", b="128k"),
>         %video(codec="libx264", b="4000k")
>     ),
>     stream
> )
> 
> flv = %ffmpeg(
>     format="flv",
>     %audio.copy,
>     %video.copy,
> )
> 
> flv_stream = ffmpeg.filter.bitstream.extract_extradata(stream)
> 
> # Send to one youtube output:
> output.youtube.live.rtmp(
>     encoder = flv,
>     flv_stream,
>     ...
> )
> 
> mpegts = %ffmpeg(
>     format="mpegts",
>     %audio.copy,
>     %video.copy,
> )
> 
> # And to a hls one:
> output.file.hls(
>   ["mpegts", mpegts],
>   stream,
>   ...
> )
> ```
